server/images.go

```diff
@@ -918,7 +918,8 @@ func ShowModelfile(model *Model) (string, error) {
 	mt.From = model.ModelPath
 
 	if model.OriginalModel != "" {
-		mt.From = model.OriginalModel
+		mp := ParseModelPath(model.OriginalModel)
```

Comment:
I was mistaken in my initial comment and this isn't quite right. `OriginalModel` is set when the model inherits from another model. At this point, the name must already be set to the canonical name with the tag otherwise it will carry forward the bug.

Therefore, the bug fix is to update `newLayer.From = mp.GetNamespaceRepository()` to `newLayer.From = mp.GetShortTagname()` in `CreateModel`